store/src/java/com/zimbra/cs/account/soap/SoapDistributionList.java

```diff
@@ -81,10 +81,30 @@
                 SoapProvisioning.getAttrs(e), prov);
         addDlm(e, getRawAttrs());
     }
-
+    
+    /**
+     * ZBUG-651: This method is modified as it is overriding the existing member/s value with zero 
+     * against zimbraMailForwardingAddress attribute.
+     * So now checking if members list is zero than it should not override.
+     * @param members This is list of members
+     * @param attrs This is list of attribute
+     * @return nothing
+     */
     private void addDlm(List <String> members, Map<String, Object> attrs) {
-        attrs.put(Provisioning.A_zimbraMailForwardingAddress,
+    	if (members.size() > 0) {
+    		attrs.put(Provisioning.A_zimbraMailForwardingAddress,
                 members.toArray(new String[members.size()]));
+    	}
+    }
+    
+    /**
+     * ZBUG-651: This method is added against the change in addDlm(), which provides blank list of members.
+     * @param attrs to put/override attribute with blank member list.
+     * @return nothing
+     */
+    private void resetDlm(Map<String, Object> attrs) {
+    	attrs.put(Provisioning.A_zimbraMailForwardingAddress,
+    			new ArrayList<String>().toArray(new String[0]));
```

Comment:
```suggestion
    			new String[0]);
```
Unless there's some reason to create an empty list just to get an empty array that I'm missing.